### PR TITLE
mediaelch: fix loading of libmediainfo

### LINKED ISSUE #159918: MediaElch does not find libmediainfo
### Describe the bug

MediaElch works correctly, except it cannot load StreamDetails. In About/Developer, it complains about not finding libmediainfo.so.0

```
MediaElch 2.8.12 - Coridian

Compiled with Qt version 5.15.3 (64 Release, GCC 10.3.0)
Using Qt version 5.15.3
System: NixOS 21.11 (Porcupine) (x86_64-little_endian-lp64)

Application dir: /nix/store/ambssjd6788n35c7yvg8kqmaqzvz71cl-mediaelch-2.8.12/bin
Settings file: /home/jalil/.config/kvibes/MediaElch.conf
Data dir: /home/jalil/.local/share/kvibes/MediaElch
Qt Translation Path: /nix/store/41zwi3a4c80vvgig5xglchkax37h04k5-qtbase-5.15.3/translations

MediaInfo Version: Unable to load libmediainfo.so.0

Default UI languages: fr-FR

Custom advancedsettings.xml: false
```

### Steps To Reproduce
1. `nix-env -i mediaelch` or add it in `/etc/nixos/configuration.nix`
2. start MediaElch and scan some files
3. go to «Stream details» tab and try to «load from files»
4. (3bis) You can also open About/Developer modal to see the message about the missing module.

### Expected behavior
MediaElch should find libmediainfo and be able to use it to get info from media.

### Screenshots

image

### Additional context
NixOS 21.11 with Gnome/Wayland

### Notify maintainers
@Stunkymonkey 
@SuperSandro2000 

### Metadata

```console
 ~  nix-shell -p nix-info --run "nix-info -m"                                       
 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.94, NixOS, 21.11 (Porcupine)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.16`
 - channels(root): `"home-manager-21.11, nixos-21.11.335665.0f316e4d72d"`
 - channels(jalil): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

```

Thanks in advance !

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#159918
@JalilArfaoui please test this.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/misc/mediaelch/default.nix [hunk 1]
```diff
@@ -38,6 +40,10 @@ mkDerivation rec {
     substituteInPlace MediaElch.pro --replace "/usr" "$out"
   '';
 
+  postInstall = ''
+    wrapProgram $out/bin/MediaElch --prefix LD_LIBRARY_PATH : "${lib.makeLibraryPath [ libmediainfo ]}"
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/mediaelch/default.nix:44]: Please use patchelf.

Comment (Mic92) [reply pkgs/applications/misc/mediaelch/default.nix:44]: You can set `runtimeDependencies` to add libraries that are opened at runtime with `dlopen`.

Comment (Stunkymonkey) [reply pkgs/applications/misc/mediaelch/default.nix:44]: with `runtimeDependencies = [ (lib.getLib libmediainfo) ];` it does not work

Comment (Stunkymonkey) [reply pkgs/applications/misc/mediaelch/default.nix:44]: with `patchelf  --add-needed ${libmediainfo}/lib/libmediainfo.so.0  $out/bin/MediaElch` it does not compile: `patchelf: not an ELF executable`

Comment (Mic92) [reply pkgs/applications/misc/mediaelch/default.nix:44]: Is this a native executable or a script?

Comment (Stunkymonkey) [reply pkgs/applications/misc/mediaelch/default.nix:44]: its a c++ project, so it has to be compiled. But due to the QT-application it gets wrapped. the last line int the wrapper calls:
```
exec -a "$0" "/nix/store/5ksgxdrgyiaxj8ijcvfnzvbsidmi2kfw-mediaelch-2.8.14/bin/.MediaElch-wrapped"  "$@"
```

Comment (SuperSandro2000) [reply pkgs/applications/misc/mediaelch/default.nix:44]: Then it should at least use qtWrapperArgs or we double wrap which should be avoided https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/qt.section.md#locating-runtime-dependencies-qt-runtime-dependencies

Comment (Stunkymonkey) [reply pkgs/applications/misc/mediaelch/default.nix]: ok thanks for the suggestions. works now. better this way?

Comment (SuperSandro2000) [reply pkgs/applications/misc/mediaelch/default.nix]: The file is loaded here https://github.com/Komet/MediaElch/blob/v2.8.14/src/data/MediaInfoFile.cpp#L23 somehow. Could we patch that instead?

Comment (Stunkymonkey) [reply pkgs/applications/misc/mediaelch/default.nix]: not sure how to patch this. could you please provide an example?

**Comment by Mic92** [comment]
I would say this is good enough :) Patching has other disadvantages.

**Comment by JalilArfaoui** [comment]
Is there something to do manually to have it available in 21.11 ?

**Comment by Stunkymonkey** [comment]
if #160508 gets merged, the fix is applied in 21.11 as well.

**Pull request merged**